panic: runtime error: index out of range at class_parser.go:196

Certain projects cause goplantuml to crash with the following error:

```
panic: runtime error: index out of range

goroutine 1 [running]:
github.com/jfeliu007/goplantuml/parser.(*ClassParser).handleGenDecl(0xc00004c140, 0xc00004ce80)
        /Users/justinw/go/src/github.com/jfeliu007/goplantuml/parser/class_parser.go:196 +0x13f3
github.com/jfeliu007/goplantuml/parser.(*ClassParser).parseFileDeclarations(0xc00004c140, 0x1133f40, 0xc00004ce80)
        /Users/justinw/go/src/github.com/jfeliu007/goplantuml/parser/class_parser.go:162 +0x81
github.com/jfeliu007/goplantuml/parser.(*ClassParser).parsePackage(0xc00004c140, 0x11331c0, 0xc000011590)
        /Users/justinw/go/src/github.com/jfeliu007/goplantuml/parser/class_parser.go:132 +0x207
github.com/jfeliu007/goplantuml/parser.(*ClassParser).parseDirectory(0xc00004c140, 0xc0000b8f00, 0x47, 0x47, 0x120e380)
        /Users/justinw/go/src/github.com/jfeliu007/goplantuml/parser/class_parser.go:153 +0xfa
github.com/jfeliu007/goplantuml/parser.NewClassDiagram.func1(0xc0000b8f00, 0x47, 0x1134cc0, 0xc00010dba0, 0x0, 0x0, 0x0, 0xc000125ab0)
        /Users/justinw/go/src/github.com/jfeliu007/goplantuml/parser/class_parser.go:89 +0x105
path/filepath.walk(0xc0000b8f00, 0x47, 0x1134cc0, 0xc00010dba0, 0xc00000c060, 0x0, 0x0)
        /usr/local/go/src/path/filepath/path.go:362 +0xe3
path/filepath.walk(0xc0000bc3c0, 0x40, 0x1134cc0, 0xc00010dad0, 0xc00000c060, 0x0, 0x0)
        /usr/local/go/src/path/filepath/path.go:382 +0x300
path/filepath.walk(0xc00001a180, 0x3c, 0x1134cc0, 0xc0000925b0, 0xc00000c060, 0x0, 0x20)
        /usr/local/go/src/path/filepath/path.go:382 +0x300
path/filepath.Walk(0xc00001a180, 0x3c, 0xc00000c060, 0x10a1d28, 0x0)
        /usr/local/go/src/path/filepath/path.go:404 +0xff
github.com/jfeliu007/goplantuml/parser.NewClassDiagram(0xc0000420c0, 0x1, 0x1, 0x120daa0, 0x0, 0x0, 0x1, 0xc0000420b0, 0xc00007ef20, 0x1010be4)
        /Users/justinw/go/src/github.com/jfeliu007/goplantuml/parser/class_parser.go:78 +0x424
main.main()
        /Users/justinw/go/src/github.com/jfeliu007/goplantuml/cmd/goplantuml/main.go:34 +0x228
```

--- a/parser/class_parser.go
+++ b/parser/class_parser.go
@@ -204,7 +204,10 @@ func (p *ClassParser) handleFuncDecl(decl *ast.FuncDecl) {
 }
 
 func (p *ClassParser) handleGenDecl(decl *ast.GenDecl) {
-
+	if decl.Specs == nil || len(decl.Specs) < 1 {
+		//This might be a type of General Declaration we do not know how to handle.
+		return
+	}
 	spec := decl.Specs[0]
 	var typeName string
 	var alias *Alias

--- a/parser/class_parser_test.go
+++ b/parser/class_parser_test.go
@@ -901,3 +901,16 @@ hide methods
 		})
 	}
 }
+
+func TestHandleGenDecl(t *testing.T) {
+	parser := getEmptyParser("main")
+	defer func() {
+		if r := recover(); r != nil {
+			t.Errorf("TestHandleGenDecl: Expected no panic in this function when the Specs are empty or nil.")
+		}
+	}()
+	parser.handleGenDecl(&ast.GenDecl{})
+	parser.handleGenDecl(&ast.GenDecl{
+		Specs: []ast.Spec{},
+	})
+}

